deepchecks/vision/segmentation_data.py

```diff
@@ -0,0 +1,193 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing the SegmentationData class and its functions."""
+from abc import abstractmethod
+from typing import List, Sequence
+
+import torch
+
+from deepchecks.core.errors import DeepchecksNotImplementedError, ValidationError
+from deepchecks.vision.vision_data import TaskType, VisionData
+
+
+class SegmentationData(VisionData):
+    """The SegmentationData class is used to load and preprocess data for a semantic segmentation task.
+
+    It is a subclass of the VisionData class. The SegmentationData class contains additional data and general
+    methods intended for easily accessing metadata relevant for validating a computer vision semantic segmentation ML
+    models.
+    """
+
+    @property
+    def task_type(self) -> TaskType:
+        """Return the task type (semantic_segmentation)."""
+        return TaskType.SEMANTIC_SEGMENTATION
+
+    @abstractmethod
+    def batch_to_labels(self, batch) -> List[torch.Tensor]:
+        """Extract the labels from a batch of data.
+
+        Parameters
+        ----------
+        batch : torch.Tensor
+            The batch of data.
+
+        Returns
+        -------
+        List[torch.Tensor]
+            List of torch.Tensors. Each tensor should be of same width and height of corresponding image, and its values
+            are the true class_ids of the corresponding pixels in that image.
+            Note that the tensor should 2D, as the number of channels on the original image are irrelevant to the class.
```

Comment:
Not the clearest sentence (suggesting to improve it, but really not a must).
Maybe would be clearer to just state the shape (H, W) or something like that

Reply:
:+1: